### DISCUSSION TITLE
How to handle strict adapter with test?

### DISCUSSION THREAD

**Discussion by sethhorsley** [jnunemaker/flipper#807] [General]
this is related to: #763

What is the expected way to handle the strict adapter in tests?

Should I add all the flipper features in a fixture or in an initializer file?

It mentions this in the docs:

> Adding feature flags to your app is as simple as adding conditionals
...
> ```ruby
> <% if Flipper.enabled? :new_feature, current_user %>
>  <%= link_to "New Feature", new_feature_path %>
> <% end %>
> ```

https://www.flippercloud.io/docs/guides/launch#add-feature-flags-to-your-app

It makes it seem like just adding the conditionals like `Flipper.enabled? :new_feature` is all you need to add a flipper feature, but in tests this warns:

```bash
Could not find feature "deep_search". Call `Flipper.add("deep_search")` to create it.
```

**Comment by bkeepers** [comment]
Hey @iseth, this is a good question and something we're still personally experimenting with since the strict adapter is so new. Our [testing docs](https://www.flippercloud.io/docs/testing) actually recommend overwriting Flipper's `config.default`, which does not use the Strict adapter at all. This is the setup we're currently using in our apps.

However, I would like to move to a setup where we're using the Strict adapter in tests, with it configured to `:raise`, and being more explicit about what features are in the code base and their state. With that, I would probably add some test-setup to add the features that are expected in the tests:

```ruby
class ActiveSupport::TestCase
  setup do
    Flipper.instance = nil
    Flipper.add :feature_1
    Flipper.add :feature_2
  end
end
```

**Reply by sethhorsley** [reply]
awesome thanks for pointing that out.

**Comment by rnestler** [comment]
We have the following in `config/initializers/flipper.rb`:

```ruby
begin
  if ActiveRecord::Base.connection.data_source_exists?('flipper_features')
    Flipper.add(:realty_import)
    Flipper.add(:portfolio)
  end
rescue ActiveRecord::NoDatabaseError
  Rails.logger.error('DB doesn\'t exist yet')
end
```

This adds the features at boot, but needs a workaround in case that there isn't a DB available yet (first time setup).
